### DISCUSSION TITLE
Is there a form data argument for setting the file output name?

### DISCUSSION THREAD

**Discussion by Peetzer** [gotenberg/gotenberg#475] [Q&A] [answered]
I use this, in combination with the httpClient in angular,

const formData: any = new FormData();
formData.append('resultFilename', `test.pdf`);

then a simple post to the gotenberg docker but a dynamic uuid is used as filename.

**Comment by gulien** [comment, marked as answer]
Hello @Peetzer,

See https://gotenberg.dev/docs/modules/api#output-filename-multipart

**Reply by Peetzer** [reply]
Thx, for those who wonder,

return this.httpClient.post(url, formData, {
      headers: new HttpHeaders({
        'Gotenberg-Output-Filename': `test`
      }),
      responseType: 'blob',
      observe: 'response'
    });

haven't tried but probably not with extraHttpHeaders?

**Comment by gulien** [comment]
`extraHttpHeaders` is a form field for the `Chromium` module, so indeed, not related to your need 😄
